arho_feature_template/core/create_geotiff.py

```diff
@@ -0,0 +1,140 @@
+import os
+
+from osgeo import gdal
+from qgis.core import QgsMapRendererParallelJob, QgsMapSettings, QgsMapSettingsUtils
+from qgis.PyQt.QtCore import QSize
+from qgis.PyQt.QtGui import QColor
+from qgis.PyQt.QtWidgets import QFileDialog
+from qgis.utils import iface
+
+from arho_feature_template.project.layers.plan_layers import (
+    LandUseAreaLayer,
+    LandUsePointLayer,
+    LineLayer,
+    OtherAreaLayer,
+    OtherPointLayer,
+    PlanLayer,
+)
+from arho_feature_template.utils.misc_utils import get_active_plan_id, use_wait_cursor
+
+
+class CreateGeoTiff:
+    def __init__(self, desired_pixel_size=0.5):
+        """Initialize the CreateGeoTiff class and fetch the required data."""
+        self.desired_pixel_size = desired_pixel_size
+        self.plan_layer = PlanLayer.get_from_project()
+
+        if not self.plan_layer:
+            iface.messageBar().pushWarning("", "Ei aktiivista kaavaa.")
+            return
+
+        self.feature = PlanLayer.get_feature_by_id(get_active_plan_id(), no_geometries=False)
+
+        if not self.feature:
+            iface.messageBar().pushWarning("", "Ei aktiivista kaavaa.")
+            return
+
+    def select_output_file(self):
+        """Opens a file dialog for the user to select the output GeoTIFF file."""
+        geotiff_path, _ = QFileDialog.getSaveFileName(
+            None, "Määritä GeoTIFF tallennuspolku", "", "GeoTIFF Files (*.tif)"
+        )
+        if not geotiff_path:
+            iface.messageBar().pushWarning("", "Tallennuspolkua ei määritetty.")
+            return None
+
+        self.create_geotiff(geotiff_path)
+
+    @use_wait_cursor
+    def create_geotiff(self, geotiff_path):
+        """Creates a GeoTIFF from the active plan layer and feature."""
+        if not self.plan_layer or not self.feature:
+            return
+
+        # Set buffered bounding box
+        bbox = self.feature.geometry().boundingBox()
+        buffer_percentage = 0.1
+        buffer_x = bbox.width() * buffer_percentage
+        buffer_y = bbox.height() * buffer_percentage
+        buffered_bbox = bbox.buffered(max(buffer_x, buffer_y))
+
+        # Rendering settings
+        settings = QgsMapSettings()
+        settings.setLayers(
+            [
+                LandUseAreaLayer.get_from_project(),
+                OtherAreaLayer.get_from_project(),
+                LandUsePointLayer.get_from_project(),
+                LineLayer.get_from_project(),
+                OtherPointLayer.get_from_project(),
+                PlanLayer.get_from_project(),
+            ]
+        )
+        settings.setBackgroundColor(QColor(255, 255, 255))
+        settings.setExtent(buffered_bbox)
```

Comment:
Todennäköisesti tästä ei tule rasteria, jonka pikselikoko olisi täsmälleen 0.5m/px.
Extent pitää asettaa siten, että alueen leveys ja korkeus on jaollinen tolla 0.5:llä.

```suggestion
        width = int(buffered_bbox.width()/self.desired_pixel_size)*self.desired_pixel_size
        height = int(buffered_bbox.height()/self.desired_pixel_size)*self.desired_pixel_size
        buffered_bbox.setXMinimum(int(buffered_bbox.xMinimum()))
        buffered_bbox.setYMinimum(int(buffered_bbox.yMinimum()))
        buffered_bbox.setXMaximum(buffered_bbox.xMinimum() + width)
        buffered_bbox.setYMaximum(buffered_bbox.yMinimum() + height)
        settings.setExtent(buffered_bbox)
```